lib/cn/kvset.c

```diff
@@ -1611,19 +1592,8 @@ kvset_pfx_lookup(
      * subsequent kblock.  It is thus safe to stop looking at
      * this kvset once there's a bloom miss.
      */
-    if (kblk->kb_blm_desc.bd_n_pages) {
-        const u64 lookup = kblk->kb_cn_bloom_lookup;
-        bool      hit = true;
-
-        if (HSE_LIKELY(lookup == BLOOM_LOOKUP_MCACHE)) {
-            hit = bloom_reader_buffer_lookup(&kblk->kb_blm_desc, kblk->kb_blm_pages, kt);
-        } else if (lookup == BLOOM_LOOKUP_BUFFER) {
-            hit = bloom_reader_buffer_lookup(&kblk->kb_blm_desc, kblk->kb_blm_pages, kt);
-        }
-
-        if (!hit)
-            goto done;
-    }
```

Comment:
Here we called bloom_reader_buffer_lookup() when I expect we intended to call bloom_reader_mcache_lookup() if (lookup == mcache).  But since cn_bloom_lookup was being initialized to zero we never called either function.